cmd/jujud/agent/machine/manifolds.go

```diff
@@ -542,16 +538,17 @@ func commonManifolds(config ManifoldsConfig) dependency.Manifolds {
 			Logger:            loggo.GetLogger("juju.worker.migrationminion"),
 		}),
 
-		// We run clock updaters for every controller machine to
-		// ensure the lease clock is updated monotonically and at a
-		// rate no faster than real time.
+		// We start clock updaters for every controller machine to ensure the
+		// lease clock is updated monotonically and at a rate no faster than
+		// real time.
+		// In practice, dependency on the Raft forwarder means that this worker
+		// will only ever be running on the Raft leader node.
```

Comment:
I'm concerned that this is 'accidental happiness' vs intentional. Maybe it is just the wording.

```
The global clock updater is responsible for maintaining a monotonically increasing clock that is no faster than real time. We depend on the raft forwarder to negotiate this information, so this updater only runs on the Raft leader node.
```

Reply:
Indeed. I looked at the commit, and I *think* the side-effect of turning it into a singleton was unintentional. I did however verify that removing and adding controller nodes keeps the worker running *somewhere* all the time. 

It seems to work correctly. Lease logic is always relative to the FSM clock, which is synced to followers, so there is a unified concept of time across controllers as it concerns leases.

Reply:
As we discussed, I am:
- Propagating the out-of-sync error back up to the caller and;
- logging at warning level instead of error for FSM operations.